### PR TITLE
List refactor and a bunch of new queries

### LINKED ISSUE #17: More Needed List Queries
Here are a bunch of more list queries we need to support:
* [x] [iwbacklinks](https://www.mediawiki.org/wiki/API:Iwbacklinks)
* [x] [logevents](https://www.mediawiki.org/wiki/API:Logevents) Support all events, and all actions.
* [x] [langbacklinks](https://www.mediawiki.org/wiki/API:Langbacklinks)
* [x] [imageusage](https://www.mediawiki.org/wiki/API:Imageusage)
* [x] [exturlusage](https://www.mediawiki.org/wiki/API:Exturlusage)
* [x] [deletedrevs](https://www.mediawiki.org/wiki/API:Deletedrevs)
* [x] [recentchanges](https://www.mediawiki.org/wiki/API:RecentChanges)
* [x] [querypage](https://www.mediawiki.org/wiki/API:Querypage)
* [x] [prefixsearch](https://www.mediawiki.org/wiki/API:Prefixsearch)
* [x] [tags](https://www.mediawiki.org/wiki/API:Tags)

### PR DESCRIPTION
**Pull request by elifoster**
This PR closes #17.

It completely refactors the list module to be more like Meta and Properties.

It adds support for all APIs described in #17.

@xbony2 Could you review this?

### CODE REVIEW

File: lib/mediawiki/query/lists/all.rb [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        # @see https://www.mediawiki.org/wiki/API:Allusers MediaWiki Allusers
+        #   API Docs
+        # @since 0.8.0
+        # @return [Hash] A hash of all users, names are keys, IDs are values.
+        def get_all_users(group = nil, limit = 500)
+          params = {
+            action: 'query',
+            list: 'allusers',
+            aulimit: get_limited(limit)
+          }
```
Comment (xbony2) [inline lib/mediawiki/query/lists/all.rb]: new line here sire :3

Comment (elifoster) [inline lib/mediawiki/query/lists/all.rb]: Any reason why? That isn't the existing style used throughout the rest of the project.

File: lib/mediawiki/query/lists/users.rb [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+          end
+
+          # %Y: Year including century
+          # %m: Month num
+          # %d: day of month
+          # %T: Time as HH:MM:SS
+          timeformat = '%Y-%m-%dT%T'
+          time = DateTime.strptime(time, timeformat)
+
+          time
```
Comment (xbony2) [inline lib/mediawiki/query/lists/users.rb]: hmm, is that statement really needed?

File: lib/mediawiki/query/lists/all.rb (hunk 1 above)
Comment (xbony2) [inline lib/mediawiki/query/lists/all.rb]: Yeah it is. A new line between the hash literal and the next statement?

Comment (elifoster) [inline lib/mediawiki/query/lists/all.rb]: Not when the next statement is adding to the hash based on conditional statements.

**Comment by elifoster** [comment]
@xbony2 How about now that I've removed that pointless declaration of the time variable in get_registration_time?

**Comment by xbony2** [comment]
Seems good :3

**Comment by xbony2** [comment]
So, are you going to merge this?

**Pull request merged**